### PR TITLE
fix(react-tabs): ignore ref for tab reserved space content slot

### PR DESCRIPTION
**Pull request by dmytrokirpa**
## Previous Behavior

Tab content slot calls ref with both content and reserved space content elements

## New Behavior

Tab content slot calls ref only with the content element and ignores ref for the reserved space content elements

## Related Issue(s)

- Fixes [#30902](https://github.com/microsoft/fluentui/issues/30902)

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31775/perf-test-react-components/Avatar-mount_base.html">659</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31775/perf-test-react-components/Avatar-mount.html">621</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31775/perf-test-react-components/Button-mount_base.html">294</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31775/perf-test-react-components/Button-mount.html">295</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31775/perf-test-react-components/Field-mount_base.html">1159</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31775/perf-test-react-components/Field-mount.html">1106</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31775/perf-test-react-components/FluentProvider-mount_base.html">712</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31775/perf-test-react-components/FluentProvider-mount.html">695</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31775/perf-test-react-components/FluentProviderWithTheme-mount_base.html">85</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31775/perf-test-react-components/FluentProviderWithTheme-mount.html">97</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31775/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">26</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31775/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">35</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31775/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">74</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31775/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">76</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31775/perf-test-react-components/MakeStyles-mount_base.html">836</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31775/perf-test-react-components/MakeStyles-mount.html">839</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31775/perf-test-react-components/Persona-mount_base.html">1767</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31775/perf-test-react-components/Persona-mount.html">1719</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31775/perf-test-react-components/SpinButton-mount_base.html">1429</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31775/perf-test-react-components/SpinButton-mount.html">1397</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31775/perf-test-react-components/SwatchPicker-mount_base.html">1630</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31775/perf-test-react-components/SwatchPicker-mount.html">1668</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
| :---------------- | -----------------------: | ----: | ---------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ReactComponents.fixture.js'>react-components: entire library</abbr> | `1.094 MB`<br />`270.606 kB` | `1.094 MB`<br />`270.659 kB` | `45 B` <br />`53 B` |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `69.141 kB`<br />`20.157 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `211.741 kB`<br />`60.957 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `44.442 kB`<br />`14.607 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `8.39 kB`<br />`2.64 kB` |
| <samp>react-timepicker-compat</samp> <br /> <abbr title='bundle-size/TimePicker.fixture.js'>TimePicker</abbr> | `104.384 kB`<br />`34.782 kB` |
</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/5d1e78ec7969cd393420404e08df66f5dee5a145'>5d1e78ec7969cd393420404e08df66f5dee5a145</a></sub>
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-tabs-c6386b10-6c09-4856-84a4-c5519c849fc3.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-tabs-c6386b10-6c09-4856-84a4-c5519c849fc3.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31775&commitId=f221b499bb26433e3f40546464af1a35201df187&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.size+inactive+badge.normal.chromium.png|0| Removed|

</details>


  <details><summary><b>Card Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Card Converged.appearance - High Contrast.normal.chromium.png|0| Removed|

</details>


  <details><summary><b>Card Converged - Interactive</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Card Converged - Interactive.appearance interactive - Subtle - High Contrast.click.chromium.png|0| Added|
|Card Converged - Interactive.appearance interactive - Subtle - High Contrast.hover.chromium.png|0| Added|
|Card Converged - Interactive.appearance interactive - Subtle - High Contrast.normal.chromium.png|0| Added|

</details>


  <details><summary><b>Image Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Image Converged.Image Shape.normal.chromium.png|0| Added|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Review by marcosmoura — APPROVED** [review]

**Comment by layershifter** [comment]
@dmytrokirpa FYI `teams-prg` does not own `react-tabs`, I created a PR to update ownership.

File: packages/react-components/react-tabs/library/src/components/Tab/useTab.ts [hunk 2]
```diff
@@ -79,7 +82,7 @@ export const useTab_unstable = (props: TabProps, ref: React.Ref<HTMLElement>): T
     icon: iconSlot,
     iconOnly,
     content: contentSlot,
-    contentReservedSpace: slot.optional(content, {
+    contentReservedSpace: slot.optional(contentReservedSpace, {
```
Comment (layershifter) [inline packages/react-components/react-tabs/library/src/components/Tab/useTab.ts:85]: _5 cents_:
I am not a big fun of this design: it should be two separate slots (`content` + `contentReservedSpace`) (if it's really needed), so they could be targeted separately. Anyway, it's too late to complain about it.

**Review by layershifter — APPROVED** [review]

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Review by GeoffCoxMSFT — APPROVED** [review]

**Pull request merged**